### ISSUE TITLE
pip install ./diff-gaussian-rasterization failed

### ISSUE THREAD

**Issue by JiewenGuan** [3DTopia/LGM#12]
os:windows 11
python: 3.11
pip list:
Package            Version
------------------ ------------
certifi            2022.12.7
charset-normalizer 2.1.1
filelock           3.9.0
fsspec             2023.4.0
idna               3.4
Jinja2             3.1.2
MarkupSafe         2.1.3
mpmath             1.3.0
networkx           3.2.1
numpy              1.26.3
pillow             10.2.0
pip                24.0
requests           2.28.1
setuptools         65.5.0
sympy              1.12
torch              2.1.0+cu118
torchaudio         2.1.0+cu118
torchvision        0.16.0+cu118
typing_extensions  4.8.0
urllib3            1.26.13
xformers           0.0.24+cu118

I was following the installation guide in the readme.md up to the 
"pip install ./diff-gaussian-rasterization"
then I encountered an error:
`
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Processing c:\users\heiye\repo\experiments\lgm\diff-gaussian-rasterization
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "C:\Users\heiye\repo\experiments\LGM\env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\heiye\repo\experiments\LGM\env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\heiye\repo\experiments\LGM\env\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\heiye\AppData\Local\Temp\pip-build-env-lgvsjz1o\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\heiye\AppData\Local\Temp\pip-build-env-lgvsjz1o\overlay\Lib\site-packages\setuptools\build_meta.py", line 295, in _get_build_requires       
          self.run_setup()
        File "C:\Users\heiye\AppData\Local\Temp\pip-build-env-lgvsjz1o\overlay\Lib\site-packages\setuptools\build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\heiye\AppData\Local\Temp\pip-build-env-lgvsjz1o\overlay\Lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 13, in <module>
      ModuleNotFoundError: No module named 'torch'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
`
it says no module named torch while I have torch installed.
how can I fix it? thank you

**Comment by ashawkey** [comment]
@JiewenGuan Hi, can you verify torch installation? like importing it and check cuda availability.

**Comment by JiewenGuan** [comment]
I got it running eventually, my problem is caused by python 3.12 as the xformers doesn't support 3.12 for now. later I got it running fine on a fresh wsl2 ubuntu 22.04 with python 3.11 and the pip list like this:
Package                Version
---------------------- -------------
blinker                1.4
command-not-found      0.3
cryptography           3.4.8
dbus-python            1.2.18
distlib                0.3.4
distro                 1.7.0
distro-info            1.1+ubuntu0.2
filelock               3.6.0
httplib2               0.20.2
importlib-metadata     4.6.4
jeepney                0.7.1
keyring                23.5.0
launchpadlib           1.10.16
lazr.restfulclient     0.14.4
lazr.uri               1.0.6
more-itertools         8.10.0
netifaces              0.11.0
oauthlib               3.2.0
pip                    22.0.2
platformdirs           2.5.1
PyGObject              3.42.1
PyJWT                  2.3.0
pyparsing              2.4.7
python-apt             2.4.0+ubuntu2
PyYAML                 5.4.1
SecretStorage          3.3.1
setuptools             59.6.0
six                    1.16.0
systemd-python         234
ubuntu-advantage-tools 8001
ufw                    0.36.1
unattended-upgrades    0.1
virtualenv             20.13.0+ds
wadllib                1.3.6
wheel                  0.37.1
zipp                   1.0.0

also it doesn't work with the default python installation of ubuntu 22.04, which is 3.10.12. great project, thank you.

**Comment by JiewenGuan** [comment]
I got it running eventually, my problem is caused by python 3.12 as the xformers doesn't support 3.12 for now. later I got it running fine on a fresh wsl2 ubuntu 22.04 with python 3.11 and the pip list like this:
Package                     Version
--------------------------- -------------
accelerate                  0.27.0
aiofiles                    23.2.1
aiohttp                     3.9.3
aiosignal                   1.3.1
altair                      5.2.0
annotated-types             0.6.0
anyio                       4.2.0
asyncer                     0.0.4
attrs                       23.2.0
blinker                     1.4
certifi                     2022.12.7
charset-normalizer          2.1.1
click                       8.1.7
colorama                    0.4.6
coloredlogs                 15.0.1
command-not-found           0.3
contourpy                   1.2.0
cryptography                3.4.8
cycler                      0.12.1
dataclasses-json            0.6.4
dbus-python                 1.2.18
dearpygui                   1.10.1
Deprecated                  1.2.14
diff-gaussian-rasterization 0.0.0
diffusers                   0.26.2
distlib                     0.3.4
distro                      1.7.0
distro-info                 1.1+ubuntu0.2
docstring-parser            0.15
einops                      0.7.0
executing                   2.0.1
fastapi                     0.109.2
ffmpy                       0.3.1
filelock                    3.6.0
filetype                    1.2.0
flatbuffers                 23.5.26
fonttools                   4.48.1
frozenlist                  1.4.1
fsspec                      2024.2.0
gradio                      4.18.0
gradio_client               0.10.0
h11                         0.14.0
httpcore                    1.0.2
httplib2                    0.20.2
httpx                       0.26.0
huggingface-hub             0.20.3
humanfriendly               10.0
idna                        3.4
imageio                     2.33.1
imageio-ffmpeg              0.4.9
importlib-metadata          4.6.4
importlib-resources         6.1.1
jeepney                     0.7.1
Jinja2                      3.1.2
joblib                      1.3.2
jsonschema                  4.21.1
jsonschema-specifications   2023.12.1
keyring                     23.5.0
kiui                        0.2.3
kiwisolver                  1.4.5
launchpadlib                1.10.16
lazr.restfulclient          0.14.4
lazr.uri                    1.0.6
lazy_loader                 0.3
llvmlite                    0.42.0
lpips                       0.1.4
markdown-it-py              3.0.0
MarkupSafe                  2.1.3
marshmallow                 3.20.2
matplotlib                  3.8.2
mdurl                       0.1.2
more-itertools              8.10.0
mpmath                      1.3.0
multidict                   6.0.5
mypy-extensions             1.0.0
netifaces                   0.11.0
networkx                    3.2.1
numba                       0.59.0
numpy                       1.26.3
nvdiffrast                  0.3.1
nvidia-cublas-cu11          11.11.3.6
nvidia-cuda-cupti-cu11      11.8.87
nvidia-cuda-nvrtc-cu11      11.8.89
nvidia-cuda-runtime-cu11    11.8.89
nvidia-cudnn-cu11           8.7.0.84
nvidia-cufft-cu11           10.9.0.58
nvidia-curand-cu11          10.3.0.86
nvidia-cusolver-cu11        11.4.1.48
nvidia-cusparse-cu11        11.7.5.86
nvidia-nccl-cu11            2.19.3
nvidia-nvtx-cu11            11.8.86
oauthlib                    3.2.0
objprint                    0.2.3
onnxruntime                 1.17.0
onnxruntime-gpu             1.17.0
opencv-python-headless      4.9.0.80
orjson                      3.9.13
packaging                   23.2
pandas                      2.2.0
pillow                      10.2.0
pip                         22.0.2
platformdirs                2.5.1
plyfile                     1.0.3
pooch                       1.8.0
protobuf                    4.25.2
psutil                      5.9.8
pydantic                    2.6.1
pydantic_core               2.16.2
pydub                       0.25.1
pygltflib                   1.16.1
Pygments                    2.17.2
PyGObject                   3.42.1
PyJWT                       2.3.0
PyMatting                   1.1.12
pyparsing                   2.4.7
python-apt                  2.4.0+ubuntu2
python-dateutil             2.8.2
python-multipart            0.0.9
pytz                        2024.1
PyYAML                      5.4.1
referencing                 0.33.0
regex                       2023.12.25
rembg                       2.0.54
requests                    2.28.1
rich                        13.7.0
roma                        1.4.2
rpds-py                     0.17.1
ruff                        0.2.1
safetensors                 0.4.2
scikit-image                0.22.0
scikit-learn                1.4.0
scipy                       1.12.0
SecretStorage               3.3.1
semantic-version            2.10.0
setuptools                  59.6.0
shellingham                 1.5.4
shtab                       1.6.5
six                         1.16.0
sniffio                     1.3.0
starlette                   0.36.3
sympy                       1.12
systemd-python              234
threadpoolctl               3.2.0
tifffile                    2024.1.30
tokenizers                  0.15.1
tomlkit                     0.12.0
toolz                       0.12.1
torch                       2.2.0+cu118
torchaudio                  2.2.0+cu118
torchvision                 0.17.0+cu118
tqdm                        4.66.2
transformers                4.37.2
trimesh                     4.1.3
triton                      2.2.0
typer                       0.9.0
typing_extensions           4.8.0
typing-inspect              0.9.0
tyro                        0.7.2
tzdata                      2024.1
ubuntu-advantage-tools      8001
ufw                         0.36.1
unattended-upgrades         0.1
urllib3                     1.26.13
uvicorn                     0.27.1
varname                     0.13.0
virtualenv                  20.13.0+ds
wadllib                     1.3.6
watchdog                    4.0.0
websockets                  11.0.3
wheel                       0.37.1
wrapt                       1.16.0
xatlas                      0.0.9
xformers                    0.0.24+cu118
yarl                        1.9.4
zipp                        1.0.0

also it doesn't work with the default python installation of ubuntu 22.04, which is 3.10.12. great project, thank you.
